enhancements/cluster-api/installing-cluster-api-components-in-ocp.md

```diff
@@ -0,0 +1,290 @@
+
+As an administrator I want to _have a CAPI-on-OpenShift architecture as minimally complicated as possible_ so that I can _extend my cluster by deploying a custom provider_.
+
+As an OpenShift engineer I want to _have a CAPI-on-OpenShift architecture as minimally complicated as possible_ so that I can _more easily maintain and extend it_.
+
+As an OpenShift engineer I want to _have a way to atomically apply a change to any provider_ so that I can avoid _payload breakages_.
+
+As an OpenShift engineer I want to _have a way to load and customize provider manifests before applying them_ so that I can _template the manifests payload with image references and other runtime tweaks_.
+
+As an OpenShift engineer I want to _have a way to load and customize provider manifests before applying them_ so that I can _future proof the system_ to be able to give users a way to selectively deploy one or more providers, even ones that are not matching the running platform.
+
+### Goals
+
+This proposal seeks to refactor OpenShift manifest generation to live in provider repositories & apply cloud platform information dynamically at runtime.  It will have succeeded when developers can update code and manifests for a Cluster API provider within one repository, and when users can selectively deploy providers at runtime.
+
+### Non-Goals
+
+This proposal does not seek to completely redesign the existing Cluster API implementation within OpenShift.
+
+We also do not seek to complete rewrite the [existing asset/manifest generation tools](https://github.com/openshift/cluster-capi-operator/tree/main/hack/assets) or switch to something like [kustomize](https://kustomize.io/). The existing tools work well enough at the moment, and will only require small changes to enable our goals. Reworking these tools or switching to another is a task for later.
+(More details on this in the [Alternatives section](#alternatives))
+
+## Proposal
+
+Considering the two challenges previously described we need to address the two following problems:
+- **Complexity** - mainly caused by the presence of two operators in the system and the number of input transformations it takes to feed them and produce the desired output
+- **Development and Payload Stability** - mainly caused by having manifests for the Cluster API providers deployed from one repository (openshift/cluster-capi-operator) and the actual code for those providers deployed from the providers repositories (openshift/cluster-api-provider-x) endangering the payload stability
+
+### Proposal 1: Improving Dev Ex + Payload Stability
+
+In order to **improve the developer experience and stability of the feature** in the system, we are proposing to **move the asset/manifest generation for the Cluster API providers as a step that happens in the provider repositories during their rebase**,
```

Comment:
It won't only be during rebase right, any time there's an update to the manifests in the repo this generation will need to run?

Reply:
Yes it is going to end up being a make target in the provider repos.
So this can be run as needed with `make manifests` (or something equivalent).
That said the main user of this is going to be the Rebase Bot, at rebase time.